File: frontend/src/domain/Unit/Unit.ts
import SiPrefixes from './SiPrefixes';
import UnitType from './UnitRype';


/**
 *  Its @readonly domain @class which contains Product characterizing
 */
class Unit {
  readonly value: number;
  readonly type: UnitType;
  readonly SiPrefix: SiPrefixes;

  /**
  *  @constructor
  *  @param {number} value
  *  @param {UnitType} type
  *  @param {SiPrefixes} SiPrefix
  */
  constructor(value: number, type: UnitType, SiPrefix: SiPrefixes) {
    this.value = value;
    this.type = type;
    this.SiPrefix = SiPrefix;
  }
}

export default Unit;

File: backend/src/main/java/com/app/hite/core/domain/unit/Unit.java
package com.app.hite.core.domain.units;

import lombok.*;


import java.lang.annotation.Annotation;

/**
 Creates a new {@literal immutable} Unit for the given {@link UnitType}, {@link SiPrefixes } and Value


 */
@Getter
@AllArgsConstructor
@ToString
@EqualsAndHashCode
@NoArgsConstructor
public class Unit  {

    private SiPrefixes siPrefix;
    private UnitType type;
    private double value;

    /**
     Creates a new {@literal immutable} Unit for the given {@link SiPrefixes }

     @param  prefix
     */
    public Unit convertToAnotherSiPrefixes(SiPrefixes prefix) {
        return new Unit(prefix, this.type, (double)(this.value * Math.pow(10, this.siPrefix.getMantissa() - prefix.getMantissa())));
    }

}
